src/ir/objc.rs

```diff
@@ -194,9 +207,55 @@ impl ObjCInterface {
             }
             CXChildVisit_Continue
         });
+
+        if interface.is_category() {
+            // If this interface is a category, we need to find the interface that this category
+            // extends.
+            let needle = format!("{}", interface.name());
+            debug!(
+                "Category {} belongs to {}, find the item",
+                interface.rust_name(),
+                needle
+            );
+            if let Some(ref mut ty) =
+                Self::get_parent_ty(ctx, real_interface_id_for_category)
+            {
+                match ty.kind_mut() {
+                    TypeKind::ObjCInterface(ref mut real_interface) => {
+                        if !real_interface.is_category() {
+                            real_interface.categories.push((
+                                interface.rust_name(),
+                                interface.template_names.clone(),
+                            ));
+                        }
+                    }
+                    _ => {}
+                }
+            }
+        }
         Some(interface)
     }
 
+    fn get_parent_ty(
+        ctx: &mut BindgenContext,
+        parent_id: Option<ItemId>,
+    ) -> Option<&mut Type> {
+        // This is pretty gross but using the ItemResolver doesn't yield a mutable reference.
+        let real_interface_item = ctx.get_item_mut(parent_id?)?;
+        let ty = real_interface_item.kind().as_type()?;
+
+        if let TypeKind::ResolvedTypeRef(item_id) = ty.kind() {
+            let real_interface_id: ItemId = item_id.into();
+            let ty = ctx
+                .get_item_mut(real_interface_id)?
```

Comment:
Why doesn't this need to be a loop? Technically there's no guarantee that there's only one layer of `ResolvedTypeRef`s.

Reply:
Hmm. Interesting. I guess I didn't realize that was a possibility... I will look into this.